docling_core/experimental/serializer/html.py

```diff
@@ -0,0 +1,922 @@
+        self,
+        *,
+        item: NodeItem,
+        doc_serializer: "BaseDocSerializer",
+        doc: DoclingDocument,
+        **kwargs,
+    ) -> SerializationResult:
+        """Fallback serializer for items not handled by other serializers."""
+        # For group items, we don't generate any markup
+        if isinstance(item, GroupItem):
+            return SerializationResult(text="")
+
+        # For other doc items, add a comment
+        return SerializationResult(
+            text=f"<!-- Unhandled item type: {item.__class__.__name__} -->"
+        )
+
+
+class HTMLDocSerializer(DocSerializer):
+    """HTML-specific document serializer."""
+
+    text_serializer: BaseTextSerializer = HTMLTextSerializer()
+    table_serializer: BaseTableSerializer = HTMLTableSerializer()
+    picture_serializer: BasePictureSerializer = HTMLPictureSerializer()
+    key_value_serializer: BaseKeyValueSerializer = HTMLKeyValueSerializer()
+    form_serializer: BaseFormSerializer = HTMLFormSerializer()
+    fallback_serializer: BaseFallbackSerializer = HTMLFallbackSerializer()
+
+    list_serializer: BaseListSerializer = HTMLListSerializer()
+    inline_serializer: BaseInlineSerializer = HTMLInlineSerializer()
+
+    params: HTMLParams = HTMLParams()
+
+    @override
+    def serialize_bold(self, text: str, **kwargs) -> str:
+        """Apply HTML-specific bold serialization."""
+        return f"<strong>{text}</strong>"
+
+    @override
+    def serialize_italic(self, text: str, **kwargs) -> str:
+        """Apply HTML-specific italic serialization."""
+        return f"<em>{text}</em>"
+
+    @override
+    def serialize_underline(self, text: str, **kwargs) -> str:
+        """Apply HTML-specific underline serialization."""
+        return f"<u>{text}</u>"
+
+    @override
+    def serialize_strikethrough(self, text: str, **kwargs) -> str:
+        """Apply HTML-specific strikethrough serialization."""
+        return f"<del>{text}</del>"
+
+    @override
+    def serialize_hyperlink(
+        self, text: str, hyperlink: Union[AnyUrl, Path], **kwargs
+    ) -> str:
+        """Apply HTML-specific hyperlink serialization."""
+        return f'<a href="{str(hyperlink)}">{text}</a>'
+
+    @override
+    def serialize_page(
+        self, parts: list[SerializationResult], **kwargs
+    ) -> SerializationResult:
+        """Serialize a page out of its parts."""
+        # Join all parts with newlines
+        body_content = "\n".join([p.text for p in parts if p.text])
+        return SerializationResult(text=f"<div class='page'>\n{body_content}\n</div>")
+
+    @override
+    def serialize_doc(
+        self, pages: dict[Optional[int], SerializationResult], **kwargs
+    ) -> SerializationResult:
+        """Serialize a document out of its pages."""
+        # Create HTML structure
+        html_parts = [
+            "<!DOCTYPE html>",
+            self._generate_head(),
+            "<body>",
+        ]
+
+        if self.params.split_page_view:
```

Comment:
for later: we should not abuse html columns for making layout. there are html standards for making this kind of views. but this can be improved later, it is only an internal detail